torch/csrc/inductor/aoti_torch/oss_proxy_executor.h

```diff
@@ -0,0 +1,99 @@
+#pragma once
+
+#include <ATen/core/dispatch/Dispatcher.h>
+#include <ATen/core/ivalue.h>
+#include <c10/macros/Export.h>
+#include <nlohmann/json.hpp>
+#include <torch/csrc/inductor/aoti_torch/c/shim.h>
+#include <torch/csrc/inductor/aoti_torch/proxy_executor.h>
+
+using json = nlohmann::json;
+
+namespace torch {
+namespace aot_inductor {
+
+enum DynamicArgType : int {
+  TensorType = 0,
+  ListTensorType = 1,
+  ListOptionalTensorType = 2,
+  IntType = 3,
+  ListIntType = 4,
+};
+
+inline bool isTensorType(DynamicArgType arg_type) {
+  return arg_type == DynamicArgType::TensorType ||
+      arg_type == DynamicArgType::ListTensorType ||
+      arg_type == DynamicArgType::ListOptionalTensorType;
+}
+
+struct DynamicArg {
+  DynamicArg(
+      int arg_index,
+      DynamicArgType arg_type,
+      int length,
+      json serialized_arg_val)
+      : arg_index(arg_index),
+        arg_type(arg_type),
+        length(length),
+        serialized_arg_val(std::move(serialized_arg_val)) {}
+  int arg_index;
+  DynamicArgType arg_type;
+  int length;
+  json serialized_arg_val;
+};
+
+struct OpKernel {
+  OpKernel(const std::string& target, const c10::OperatorHandle& op_handle)
+      : target_(target), op_handle_(op_handle) {}
+
+  std::string target_;
+  c10::OperatorHandle op_handle_;
+  std::vector<DynamicArg> dynamic_args_;
+  std::vector<DynamicArg> outputs_;
+  std::vector<c10::IValue> stack_;
+
+  int num_output_tensors() const {
```

Comment:
Why signed value?

Reply:
I just followed [these signed values](https://github.com/pytorch/pytorch/blob/f093cd40864d5c8d4ceecb72fc9828a9fcb94a4e/torch/csrc/inductor/aoti_torch/proxy_executor.h#L15-L20) 😅 not sure if those should also be changed to size_t